src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2014,7 +2011,7 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMValidatorList> const& m)
 {
     try
     {
-        assert(supportsProtocolVersion("XRPL/2.1"));
+        assert(supportsFeature(ProtocolFeature::ValidatorListPropagation));
```

Comment:
This is a fine check to have, but I'd turn it into this:

    if (!supportsFeature(ProtocolFeature::ValidatorListPropagation))
        return;

Rationale: as is, the `assert` would allow a remote server to crash a server running this changeset with asserts enabled just by sending an unexpected message.

Reply:
Good catch. Fixed, including charging a fee for the unwanted data.